cpp/src/copying/copy.cu

```diff
@@ -267,6 +268,22 @@ struct copy_if_else_functor_impl<list_view> {
   }
 };
 
+template <>
```

Comment:
Not a bug but I observe that this file contains too many struct specializations that have the same implementation. Maybe `enable_if` should be used instead of specialization.

Reply:
Approved, but still hope if this can be improved (here or in a new PR) :)

Reply:
I disagree with converting these to `enable_if`. I like how each type has its own specialization and each one calls only a single function.

Reply:
I don't like `enable_if` too. But I believe that we can simplify the code somehow. Otherwise, we still have multiple structs having the same implementation.

Reply:
Maybe using inheritance? I.e., the structs sharing common code will just inherit from the same base.

Reply:
To clarify: Above is just my suggestion, not request, which can be done outside of this PR.